### PR TITLE
Fee Transfer Tests

### PR DESCRIPTION
**Pull request by Paul4912**
Motivation: Add tests to features added previously.

Changes: Tests for fee transfer. Test that transfers correct amount of fee to correct addresses for both primary and secondary fee receiver. Also added a test for paused functions.

### CODE REVIEW

**Comment by mynameuhh** [comment]
Can you please add a description to this? Preferably using the "Motivation + changes" format

File: test/LeveragedPool/transferFees.spec.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+const amountMinted = ethers.utils.parseEther("10000")
+const feeAddress = generateRandomAddress()
+const secondFeeAddress = generateRandomAddress()
+const lastPrice = ethers.utils.parseEther(getRandomInt(99999999, 1).toString())
+const updateInterval = 200
+const frontRunningInterval = 100 // seconds
+const fee = ethers.utils.parseEther("0.1")
+const leverage = 1
+
+describe("LeveragedPool - feeTrasnfer", () => {
```
Comment (sporejack) [inline test/LeveragedPool/transferFees.spec.ts:41]:
```suggestion
describe("LeveragedPool - feeTransfer", () => {
```

File: test/LeveragedPool/transferFees.spec.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            feesPaidExpected * 0.9,
+            0.00001
+        )
+        expect(parseFloat(ethers.utils.formatEther(feesPaidSecondary))).closeTo(
+            feesPaidExpected * 0.1,
+            0.00001
+        )
+    })
+
+    describe("Test Paused Pools cannot transfer tokens", async () => {
```
Comment (sporejack) [inline test/LeveragedPool/transferFees.spec.ts:121]:
```suggestion
    context("Test Paused Pools cannot transfer tokens", async () => {
```

**Review by sporejack — CHANGES_REQUESTED** [review]

**Review by mynameuhh — CHANGES_REQUESTED** [review]
Generally we try to use a more behaviour oriented structure for test files, where we wrap tests of a related context in a `context` branch, with any number of `it`s for testing certain things within that functionality.

We as a team haven't really sat down and discussed exactly how we want our test suite to be laid out (something I will be adding to the backlog), but I would say the tests added here should be structured a bit more robustly so when we come around to refactoring it, we can reason about what is happening here a bit more easily.

**Comment by mynameuhh** [comment]
> Generally we try to use a more behaviour oriented structure for test files, where we wrap tests of a related context in a `context` branch, with any number of `it`s for testing certain things within that functionality.
> 
> We as a team haven't really sat down and discussed exactly how we want our test suite to be laid out (something I will be adding to the backlog), but I would say the tests added here should be structured a bit more robustly so when we come around to refactoring it, we can reason about what is happening here a bit more easily.

I'll add that this is something I haven't done amazingly in the past but soon we'll have specifics and enforce them

**Comment by Paul4912** [comment]
> Generally we try to use a more behaviour oriented structure for test files, where we wrap tests of a related context in a `context` branch, with any number of `it`s for testing certain things within that functionality.
> 
> We as a team haven't really sat down and discussed exactly how we want our test suite to be laid out (something I will be adding to the backlog), but I would say the tests added here should be structured a bit more robustly so when we come around to refactoring it, we can reason about what is happening here a bit more easily.

I agree i try to wrap them in context where appropriate. But i think when there is like 1 or two that kind of fit in their own category i think theres no need to group them. I think we should apply groupings when there is enough to consider a group.

**Comment by Paul4912** [comment]
> > Generally we try to use a more behaviour oriented structure for test files, where we wrap tests of a related context in a `context` branch, with any number of `it`s for testing certain things within that functionality.
> > We as a team haven't really sat down and discussed exactly how we want our test suite to be laid out (something I will be adding to the backlog), but I would say the tests added here should be structured a bit more robustly so when we come around to refactoring it, we can reason about what is happening here a bit more easily.
> 
> I agree i try to wrap them in context where appropriate. But i think when there is like 1 or two that kind of fit in their own category i think theres no need to group them. I think we should apply groupings when there is enough to consider a group.

this one in particular i can group my two tests since they are happy path test cases.

**Review by mynameuhh — APPROVED** [review]

**Review by sporejack — APPROVED** [review]

**Pull request merged**